Issue 13955 - 64 bit C ABI not followed for passing structs with mixed floating types

yebblies opened the pull request:
Check code in argtypes.c should be checking that both types can be passed in xmm registers.

https://issues.dlang.org/show_bug.cgi?id=13955

ibuclaw commented on src/argtypes.c:
```diff
@@ -429,7 +429,8 @@ TypeTuple *toArgTypes(Type *t)
                     {
-                        if (t1->ty == Tfloat64 && t2->ty == Tfloat64)
+                        if ((t1->ty == Tfloat32 || t1->ty == Tfloat64) &&
+                            (t2->ty == Tfloat32 | t2->ty == Tfloat64))
```
Is the single pipe a typo?

yebblies replied:
xor?

yebblies replied:
But yeah, fixed.

ibuclaw replied:
bitwise or... I always get my or's mixed up. :)

yebblies committed:
Fix Issue 13955 - 64 bit C ABI not followed for passing structs with mixed floating types

Check code in argtypes.c should be checking that both types can be passed in xmm registers.

ibuclaw commented:
Auto-merge toggled on

yebblies commented:
@klickverbot Yep, here it is:

https://gist.github.com/yebblies/a848bd1326b380b471e8

@ibuclaw Thanks.
